scheduler/reconcile.go

```diff
@@ -254,12 +264,33 @@ func (a *allocReconciler) Compute() *reconcileResults {
 			}
 		}
 	}
+}
 
-	return a.result
+// TODO: Can/should these be converted to functions?
```

Comment:
Yeah. I _really_ dislike that the scheduler is full of methods that mutate reconciler state rather than pure functions, because it makes it painfully easy to have invisible side-effects. (It _may_ also reduce memory allocations, so there are tradeoffs.) But that kind of sweeping change will take a big effort to ensure we're not breaking any of those side-effects!